Bugzilla/WebService/User.pm

```diff
@@ -135,6 +137,62 @@ sub create {
     return { id => $self->type('int', $user->id) };
 }
 
+sub suggest {
+    my ($self, $params) = @_;
+
+    Bugzilla->switch_to_shadow_db();
+
+    ThrowCodeError('params_required', { function => 'User.suggest', params => ['match'] })
+      unless defined $params->{match};
+
+    ThrowUserError('user_access_by_match_denied')
+      unless Bugzilla->user->id;
+
+    untaint($params->{match});
+    my $s = $params->{match};
+    trim($s);
+    return { users => [] } if length($s) < 3;
+
+    my $dbh = Bugzilla->dbh;
+    my @select = ('realname AS real_name', 'login_name AS name');
+    my $order  = 'last_seen_date DESC';
+    my $where;
+    state $have_mysql = $dbh->isa('Bugzilla::DB::Mysql');
+
+    if ($s =~ /^[:@](.+)$/s) {
+        $where = $dbh->sql_prefix_match(nickname => $1);
+    }
+    elsif ($s =~ /@/) {
+        $where = $dbh->sql_prefix_match(login_name => $s);
+    }
+    else {
+        if ($have_mysql && ( $s =~ /[[:space:]]/ || $s =~ /[^[:ascii:]]/ ) ) {
+            my $match = sprintf 'MATCH(realname) AGAINST (%s) ', $dbh->quote($s);
+            push @select, "$match AS relevance";
+            $order = 'relevance DESC';
+            $where = "$match";
+        }
+        else {
+            $where = join(' OR ',
+                $dbh->sql_prefix_match(nickname => $s),
+                $dbh->sql_prefix_match(login_name => $s),
+            );
+        }
+    }
+    $where = "($where) AND is_enabled = 1";
+
+    my $sql = "SELECT " . join(", ", @select) . " FROM profiles WHERE $where ORDER BY $order LIMIT 25";
```

Comment:
if you want to keep oracle happy you'll need to use `$dbh->sql_limit()` instead of `LIMIT`